csrc/fusion.cpp

```diff
@@ -785,6 +786,12 @@ void Fusion::aliasOutputToInput(
     output = set(output);
   }
 
+  // replays the allocation domain and contiguity of input on output
+  NVF_CHECK(output->isA<TensorView>() && input->isA<TensorView>(),
+            "aliasing output to input is only supported for TensorView");
+  auto new_domain = TransformReplay::selfAllocationReplay(output->as<TensorView>()->domain(), input->as<TensorView>()->domain());
+  output->as<TensorView>()->setAllocationDomain(new_domain->allocation(), new_domain->contiguity());
```

Comment:
when we are aliasing output to input, the two needs to have `identical` allocation domain and contiguity. (technically we only needed compatible allocation domain, but this is easier for the time being).